modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/IgniteCacheDatabaseSharedManager.java

```diff
@@ -650,14 +652,31 @@ static void checkWalArchiveSizeConfiguration(
 
             long min = memCfg.getMinWalArchiveSize();
 
-            if (min != HALF_MAX_WAL_ARCHIVE_SIZE && min > max) {
-                throw new IgniteCheckedException(String.format(
-                    "DataRegionConfiguration.minWalArchiveSize must be less than or equal to " +
-                        "DataRegionConfiguration.maxWalArchiveSize or equal to %d " +
-                        "(to be half of maxWalArchiveSize), current settings:" + U.nl() +
-                        "DataRegionConfiguration.minWalArchiveSize: %d bytes" + U.nl() +
-                        "DataRegionConfiguration.maxWalArchiveSize: %d bytes",
-                    HALF_MAX_WAL_ARCHIVE_SIZE, min, max));
+            double percentage = getDouble(IGNITE_THRESHOLD_WAL_ARCHIVE_SIZE_PERCENTAGE, -1);
+
+            if (min != HALF_MAX_WAL_ARCHIVE_SIZE) {
+                if (min > max) {
+                    throw new IgniteCheckedException(String.format(
+                        "DataRegionConfiguration.minWalArchiveSize must be less than or equal to " +
+                            "DataRegionConfiguration.maxWalArchiveSize or equal to %d " +
+                            "(to be half of maxWalArchiveSize), current settings:" + U.nl() +
+                            "DataRegionConfiguration.minWalArchiveSize: %d bytes" + U.nl() +
+                            "DataRegionConfiguration.maxWalArchiveSize: %d bytes",
+                        HALF_MAX_WAL_ARCHIVE_SIZE, min, max));
+                }
+            }
+            else if (percentage != -1) {
+                log.warning(String.format(
```

Comment:
Shouldn't we also validate that "percentage" is positive?

Reply:
In this case, 0 and a negative value does not break the logic, since we will always try to delete segments until this value.